bindings/go/goucx.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2021, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "goucx.h"
+
+static inline void complete_go_send_request(void *request,
+                                            ucs_status_t status, void *user_data)
+{
+    completeGoSendRequest(request, status, (uint64_t)user_data);
+}
+
+static inline void complete_go_recv_callback(void *request, ucs_status_t status,
+                                             const ucp_tag_recv_info_t *info, void *user_data)
+{
+    completeGoTagRecvRequest(request, status, *info, (uint64_t)user_data);
+}
+
+static inline void complete_go_error_handler(void *arg, ucp_ep_h ep, ucs_status_t status)
+{
+    completeGoErrorHandler(ep, status);
+}
+
+void set_send_callback(ucp_request_param_t *params, uint64_t callback_id)
+{
+    params->op_attr_mask |= UCP_OP_ATTR_FIELD_CALLBACK | UCP_OP_ATTR_FIELD_USER_DATA;
+    params->cb.send = complete_go_send_request;
+    params->user_data = (void *)callback_id;
+}
+
+void set_tag_recv_callback(ucp_request_param_t *params, ucp_tag_recv_info_t *info, uint64_t callback_id)
+{
+    params->op_attr_mask |= UCP_OP_ATTR_FIELD_CALLBACK | UCP_OP_ATTR_FIELD_USER_DATA;
+    params->cb.recv = complete_go_recv_callback;
+    params->recv_info.tag_info = info;
+    params->user_data = (void *)callback_id;
+}
+
+void set_error_handler(ucp_err_handler_t *err_hndl)
+{
+    err_hndl->cb = complete_go_error_handler;
+}
+
+void* get_request_ptr(ucs_status_ptr_t request)
+{
+    return UCS_STATUS_PTR(request);
+}
+
+ucs_status_t get_request_status(ucs_status_ptr_t request)
+{
+    return UCS_PTR_IS_PTR(request) ? ucp_request_check_status(UCS_STATUS_PTR(request)) : UCS_PTR_RAW_STATUS(request);
```

Comment:
can we forward constants from C to Go and implement this in go?
since it's not calling UCX C API
+ add newline